File: sdks/java/src/main/java/com/boltprediction/sdk/model/ThresholdConfigSchema.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 预警阈值配置 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class ThresholdConfigSchema {

    @JsonProperty("high_risk_threshold")
    private Integer highRiskThreshold;

    @JsonProperty("medium_risk_threshold")
    private Integer mediumRiskThreshold;

    @JsonProperty("min_normal_preload")
    private Double minNormalPreload;

    @JsonProperty("max_normal_preload")
    private Double maxNormalPreload;

    @JsonProperty("warning_deviation")
    private Double warningDeviation;

    @JsonProperty("critical_deviation")
    private Double criticalDeviation;

    @JsonProperty("auto_create_work_order_level")
    private Integer autoCreateWorkOrderLevel;

    @JsonProperty("default_upgrade_minutes")
    private Integer defaultUpgradeMinutes;

    public Integer getHighRiskThreshold() {
        return highRiskThreshold;
    }

    public void setHighRiskThreshold(Integer highRiskThreshold) {
        this.highRiskThreshold = highRiskThreshold;
    }

    public Integer getMediumRiskThreshold() {
        return mediumRiskThreshold;
    }

    public void setMediumRiskThreshold(Integer mediumRiskThreshold) {
        this.mediumRiskThreshold = mediumRiskThreshold;
    }

    public Double getMinNormalPreload() {
        return minNormalPreload;
    }

    public void setMinNormalPreload(Double minNormalPreload) {
        this.minNormalPreload = minNormalPreload;
    }

    public Double getMaxNormalPreload() {
        return maxNormalPreload;
    }

    public void setMaxNormalPreload(Double maxNormalPreload) {
        this.maxNormalPreload = maxNormalPreload;
    }

    public Double getWarningDeviation() {
        return warningDeviation;
    }

    public void setWarningDeviation(Double warningDeviation) {
        this.warningDeviation = warningDeviation;
    }

    public Double getCriticalDeviation() {
        return criticalDeviation;
    }

    public void setCriticalDeviation(Double criticalDeviation) {
        this.criticalDeviation = criticalDeviation;
    }

    public Integer getAutoCreateWorkOrderLevel() {
        return autoCreateWorkOrderLevel;
    }

    public void setAutoCreateWorkOrderLevel(Integer autoCreateWorkOrderLevel) {
        this.autoCreateWorkOrderLevel = autoCreateWorkOrderLevel;
    }

    public Integer getDefaultUpgradeMinutes() {
        return defaultUpgradeMinutes;
    }

    public void setDefaultUpgradeMinutes(Integer defaultUpgradeMinutes) {
        this.defaultUpgradeMinutes = defaultUpgradeMinutes;
    }

}

File: sdks/python/bolt_prediction_sdk/models/threshold_config_schema.py
"""
ThresholdConfigSchema 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class ThresholdConfigSchema(SDKBaseModel):
    """预警阈值配置"""

    high_risk_threshold: Optional[int] = Field(description="高风险阈值", default=3)
    medium_risk_threshold: Optional[int] = Field(description="中风险阈值", default=7)
    min_normal_preload: Optional[float] = Field(description="正常预紧力最小值", default=400)
    max_normal_preload: Optional[float] = Field(description="正常预紧力最大值", default=800)
    warning_deviation: Optional[float] = Field(description="预警偏差比例", default=0.1)
    critical_deviation: Optional[float] = Field(description="紧急偏差比例", default=0.2)
    auto_create_work_order_level: Optional[int] = Field(description="自动创建工单的最低告警级别", default=3)
    default_upgrade_minutes: Optional[int] = Field(description="默认未处理升级时间（分钟）", default=30)
